components/application-mgt/org.wso2.carbon.identity.application.mgt/src/main/java/org/wso2/carbon/identity/application/mgt/ApplicationManagementServiceImpl.java

```diff
@@ -199,9 +211,13 @@ public void updateApplication(ServiceProvider serviceProvider, String tenantDoma
 
         // invoking the listeners
         Collection<ApplicationMgtListener> listeners = ApplicationMgtListenerServiceComponent.getApplicationMgtListeners();
-        for (ApplicationMgtListener listener : listeners) {
-            if (listener.isEnable() && !listener.doPreUpdateApplication(serviceProvider, tenantDomain, username)) {
-                return;
+        if (isImportSP.get()) {
+            isImportSP.set(false);
+        } else {
+            for (ApplicationMgtListener listener : listeners) {
+                if (listener.isEnable() && !listener.doPreUpdateApplication(serviceProvider, tenantDomain, username)) {
```

Comment:
We need to throw an exception (since we return void) if a pre listener fails. Otherwise there will be no way for the callee to know if the update was success or not